README.md

```diff
@@ -74,23 +79,54 @@ from within each directory.
+## Data Modeling
+
+### Street Graph
+
+<img src='readme/basemap.jpg' width=800 alt="San Francisco basemap data" />
+
+The Shark Walk worker builds a geographic data model by accessing the San Francisco Open Data API's [Street Intersections Dataset](https://data.sfgov.org/Geographic-Locations-and-Boundaries/Street-Intersections/ctsg-7znq) and the [Streets of San Francisco Dataset](https://data.sfgov.org/Geographic-Locations-and-Boundaries/Streets-of-San-Francisco/4ugn-hf48). The worker creates a graph where each intersection is a node and each immediately-connected intersection is an edge. The worker calculates and stores the distance between the node and each edge to speed up pathfinding later down the road.
+
+The Shark Walk worker also uses the [Streets of San Francisco Dataset](https://data.sfgov.org/Geographic-Locations-and-Boundaries/Streets-of-San-Francisco/4ugn-hf48) to create a record of each point in the polylines that describe the city streets. It saves these points, along with the intersections they are connected to, as a secondary graph that is used to find the 'entry' and 'exit' nodes during [pathfinding](#pathfinding).
+
+### Crime Risk
+
+<img src='readme/riskweight.png' width=800 alt="Crime risk weight per distance" />
+
+To build a crime risk model, the worker accesses the [SFPD Incidents Dataset](https://data.sfgov.org/Public-Safety/SFPD-Incidents-from-1-January-2003/tmnf-yvry) to retrieve criminal police reports from the last six months. For each node in the street graph, the worker computes a crime risk for that node by examining police reports in the vicinity. The closer a police report is to the node, the more it contributes to the risk at that node. A report that is at the node adds a value of one to the crime risk at the node. The weight of the police report on the crime risk falls off according to the modified sigmoid function shown above. At 100m from the node, a report adds 0.5 to the crime risk weight. At 250m from the node, a police report adds less that 0.001 to the crime risk weight. The worker ignores reports outside of 250m.
+
+## Pathfinding
+
+Armed with a graph of the city's streets and a risk level at each point, Shark Walk is able to develop deterministic optimal paths between two points in the city based on the weight given to the crime risk, or in other words, how far out of one's way one is willing to go to avoid crime risk.
+
+The first task is to compute the 'entry' and 'exit' nodes, which are the first and last nodes on the graph in the journey (this is necessary since the user's origin and destination will not exactly correspond to nodes in the Shark Walk geographic data model). Shark Walk selects the closest point on the street city street to the user's origin and destination from the secondary graph created by the worker. This point is connected to two intersection nodes and one is selected by Shark Walk to be either the entry node (close to the origin) or the exit node (close to the destination).
```

Comment:
"Shark Walk selects the closest point on the **_street city street_** to the user's origin and destination from the secondary graph created by the worker."

Typo?